pkg/controllers/mcs/service_export_controller.go

```diff
@@ -204,6 +204,17 @@ func (c *ServiceExportController) enqueueReportedEpsServiceExport() {
 	}
 }
 
+func (c *ServiceExportController) waitTargetResourceSyncedOnCluster(ctx context.Context, cluster string, interval time.Duration, gvr schema.GroupVersionResource) error {
+	return wait.PollUntilContextCancel(ctx, interval, true,
+		func(_ context.Context) (done bool, err error) {
+			synced := c.InformerManager.WaitForCacheSync(cluster)
+			if synced == nil || !synced[gvr] {
+				return false, nil
+			}
+			return true, nil
+		})
+}
```

Comment:
```suggestion
// waitForResourceSyncOnCluster waits until the specified resource's cache is synced on the given cluster.
// It polls at 1 second intervals until the resource is synced or the context is cancelled.
// Returns an error if the context is cancelled before the resource is synced.
// targetResource: the GroupVersionResource to wait for cache sync.
// targetCluster: the name of the cluster to check for cache sync.
// Returns nil if the resource cache is synced, or an error if the context is cancelled.
func (c *ServiceExportController) waitForResourceSyncOnCluster(ctx context.Context, targetResource schema.GroupVersionResource, targetCluster string) error {
	return wait.PollUntilContextCancel(ctx, time.Second, true, func(_ context.Context) (done bool, err error) {
		synced := c.InformerManager.WaitForCacheSync(targetCluster)
		if synced == nil || !synced[targetResource] {
			return false, nil
		}
		return true, nil
	})
}
```
I don't usually comment on what a code block does, but I expect an exception for this controller because of the complexity.